### PR TITLE
stat: improve doc comments

### PR DESCRIPTION
**Pull request by kortschak**

### CODE REVIEW

File: covariancematrix.go [hunk 1]
```diff
@@ -72,14 +74,16 @@ func CovarianceMatrix(cov *mat64.SymDense, x mat64.Matrix, weights []float64) *m
...
 //
-// The weights must have length equal to the number of rows in
-// input data matrix x. If corr is nil, then a new matrix with appropriate size will
-// be constructed. If corr is not nil, it should have the same number of columns
-// as the input data matrix x, and it will be used as the destination for the
-// correlation data. Weights must not be negative.
+// If weights is not nil the weighted correlation of x is calculated. weights
+// must have length equal to the number of rows in input data matrix and
+// must not contain negative elements.
+// If cov is nil, a new matrix with appropriate size will be constructed.
```
Comment (vladimir-ch) [inline covariancematrix.go]: `If corr is nil ...`. Also below.

Comment (kortschak) [inline covariancematrix.go]: Thanks

File: covariancematrix.go [hunk 2]
```diff
@@ -12,15 +12,17 @@ import (
 	"github.com/gonum/matrix/mat64"
 )
 
-// CovarianceMatrix calculates a covariance matrix (also known as a
-// variance-covariance matrix) from a matrix of data, using a two-pass
-// algorithm.
+// CovarianceMatrix returns the covariance matrix (also known as a
```
Comment (vladimir-ch) [inline covariancematrix.go]: `... known as the ...`? Not sure.

File: covariancematrix.go [hunk 3]
```diff
@@ -12,15 +12,17 @@ import (
...
 //
-// The weights must have length equal to the number of rows in
-// input data matrix x. If cov is nil, then a new matrix with appropriate size will
-// be constructed. If cov is not nil, it should have the same number of columns as the
-// input data matrix x, and it will be used as the destination for the covariance
-// data. Weights must not be negative.
+// If weights is not nil the weighted covariance of x is calculated. weights
+// must have length equal to the number of rows in input data matrix and
+// must not contain negative elements.
+// If cov is nil, a new matrix with appropriate size will be constructed.
```
Comment (btracey) [inline covariancematrix.go]: Newline?

**Comment by btracey** [comment]
LGTM

**Pull request merged**